### PR TITLE
Feature/watson cpp component

### PR DESCRIPTION
**Pull request by achocron**
Main change is that the Watson component is now working on RS2 but NuGet requires assemblies compiled for several architectures. Also reduced number of .sln files and modified Puppet to use project references. Merging is blocked until VSTS definitions are updated to reflect the changes made here.

### CODE REVIEW

File: MobileCenter-Windows.sln [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 15
-VisualStudioVersion = 15.0.26228.12
+VisualStudioVersion = 15.0.26228.9
```
Comment (guperrot) [inline MobileCenter-Windows.sln]: You should update one day ;)

File: SDK/MobileCenter/WatsonRegistrationUtility/WatsonRegistrationManager.h [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿#pragma once
+
+namespace WatsonRegistrationUtility
+{
+	public ref class WatsonRegistrationManager sealed
```
Comment (guperrot) [inline SDK/MobileCenter/WatsonRegistrationUtility/WatsonRegistrationManager.h]: indentation looks wrong on that file

Comment (achocron) [inline SDK/MobileCenter/WatsonRegistrationUtility/WatsonRegistrationManager.h]: Can you elaborate? Looks fine to me

**Comment by achocron** [comment]
@guperrot What caused that error message?

File: SDK/MobileCenter/WatsonRegistrationUtility/WatsonRegistrationManager.h (hunk 2 above)
Comment (guperrot) [inline SDK/MobileCenter/WatsonRegistrationUtility/WatsonRegistrationManager.h]: Why is public keyword on same indentation level as opening curly brace? Is that a c++ convention?

**Comment by guperrot** [comment]
@achocron I reinstalled VS 2017 and after a couple clean it eventually started to work.

File: SDK/MobileCenter/WatsonRegistrationUtility/WatsonRegistrationManager.h (hunk 2 above)
Comment (achocron) [inline SDK/MobileCenter/WatsonRegistrationUtility/WatsonRegistrationManager.h]: Yep, it is.

**Comment by achocron** [comment]
@guperrot Can't remove MobileCenter-Demo-Mac.sln as it is used in one of the cake tasks (UpdateDemoDependencies in version.cake)

**Comment by guperrot** [comment]
@achocron Can we then rename the SDK one to Mac and remove UWP projects from there?

File: SDK/MobileCenter/Microsoft.Azure.Mobile/MobileCenter.cs [hunk 3]
```diff
@@ -26,7 +26,7 @@ internal MobileCenter()
         ///     Enable or disable the SDK as a whole. Updating the property propagates the value to all services that have been
         ///     started.
         /// </summary>
-        /// <remarks>
+        /// <remarks>ah
```
Comment (guperrot) [inline SDK/MobileCenter/Microsoft.Azure.Mobile/MobileCenter.cs]: typo

**Pull request merged**